### PR TITLE
ROX-25860: Use retrying kubectl wrapper in scanner V4 test suite

### PR DESCRIPTION
**Pull request by mclasmeier**
### Description

The test suite [failed]([1825905085201256448](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-stackrox-stackrox-master-ocp-4-16-merge-scanner-v4-tests/1825905085201256448)) due to an i/o timeout error while talking to the Kube API server.

This PR changes the test suite for scanner V4 to use a retrying wrapper for kubectl which is already used by the operator scripts.

### User-facing documentation

- [x] CHANGELOG is not needed
- [x] documentation changes not needed

### Testing and quality

- [x] CI results are inspected

#### Automated testing

- [x] modified existing tests

#### How I validated my change

Just CI.

### CODE REVIEW

File: scripts/retry-kubectl.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```
Comment (mclasmeier) [inline scripts/retry-kubectl.sh:1]: I actually don't know why this file isn't being shown as being renamed, since it has been moved with `git mv`.

Comment (porridge) [reply scripts/retry-kubectl.sh:1]: I think replacing the original file with a symlink confuses GH.
🤷🏻

File: tests/e2e/run-scanner-v4.bats [hunk 2]
```diff
@@ -615,13 +615,13 @@ verify_no_scannerV4_deployed() {
 
 verify_no_scannerV4_indexer_deployed() {
     local namespace=${1:-stackrox}
-    run kubectl -n "$namespace" get deployments -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}'
+    run "${ORCH_CMD}" -n "$namespace" get deployments -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}'
```
Comment (porridge) [inline tests/e2e/run-scanner-v4.bats:618]: I have no experience with BATS, but please note that the script is not exactly a drop-in replacement for bare `kubectl`. In particular:

> NOTE: Reads all of stdin. If no input should be passed to kubectl, caller should redirect stdin from /dev/null!

Same below.

**Review by porridge — APPROVED** [review]

File: tests/e2e/run-scanner-v4.bats (hunk 2 above)
Comment (mclasmeier) [reply tests/e2e/run-scanner-v4.bats]: Thanks for the reminder. I have verified that bats doesn't fiddle around with stdin.
That means, when writing

```
run cmd
```

the semantics of reading `stdin` are unchanged for `cmd`, regardless of this thing running in a bats context or not.

That being said, it seems to work in CI. But according to my tests it wouldn't work in an interactive shell. Hence I have added the `</dev/null`.

**Review by porridge — APPROVED** [review]

**Comment by porridge** [comment]
Looks like the test hung. Not sure if this is related...
Would it be possible to deduce where it hung from the corrected artifacts?

**Comment by mclasmeier** [comment]
@porridge, not obvious to me what happened there. What surprises me the most is that there hasn't been anything written to the log about this specific timeout.

**Pull request merged**